cmd/kops/update_cluster.go

```diff
@@ -144,6 +144,22 @@ func RunUpdateCluster(f *util.Factory, clusterName string, out io.Writer, c *Upd
 		}
 
 		glog.Infof("Using SSH public key: %v\n", c.SSHPublicKey)
+	} else {
+		items, err := keyStore.FindSSHPublicKeys(fi.SecretNameSSHPrimary)
+		if err != nil {
+			return fmt.Errorf("error finding SSH Key: %v", err)
+		}
+		if len(items) == 1 {
+			c.SSHPublicKey = string(items[0].Data)
```

Comment:
Closing this PR.. I no longer have ability/desire to push from DualSpark.. Let's let #1207 handle this moving forward